### PR TITLE
Add CI_SERVICE_alias environment variables to make referencing the FQDN of services fun!

### PR DESCRIPTION
**Pull request by pauldoomgov**
This addresses #7

This change injects an environment variable named `CI_SERVICE_<alias>` for each service and sets the value to the fully qualified domain name of the service. Here is an example snippet of `.gitlab-ci.yml` showing its use:

~~~yaml
echo-test-job:
  stage: test
  image: public.ecr.aws/docker/library/ubuntu:jammy
  services:
  - name: ghcr.io/mendhak/http-https-echo:latest
    alias: echo
  script:
  - curl http://${CI_SERVICE_echo}:8080
~~~

The old method to get to the same outcome required cobbling the variable together from 5 other variables.

Instead of attempting to hack the GitLab generated command list, we just calculate and inject the CI_SERVICE_alias variables in `prepare.sh` using a modified CloudFoundry manifest. It might be hideous, but since the need to calculate a service name dynamically is not present in Docker, Kubernetes, and other executors with ephemeral local network namespaces, it seems right.

### CODE REVIEW

**Comment by pauldoomgov** [comment]
Pipeline proof of life for folks who want "evidence" - https://gitlab.com/pauldoomgov/hyperloop-traffic-jam/-/pipelines/1399063314

File: runner/cf-driver/base.sh [hunk 1]
```diff
@@ -10,12 +10,12 @@ CONTAINER_ID="glrw-r$CUSTOM_ENV_CI_RUNNER_ID-p$CUSTOM_ENV_CI_PROJECT_ID-c$CUSTOM
...
 if [ -z "$DEFAULT_JOB_IMAGE" ]; then
     DEFAULT_JOB_IMAGE="ubuntu:latest"
-    echo "WARNING: DEFAULT_JOB_IMAGE not set! Falling back to ${DEFAULT_JOB_IMAGE}"
+    echo "WARNING: DEFAULT_JOB_IMAGE not set! Falling back to ${DEFAULT_JOB_IMAGE}" 1>&2
 fi
 
 # Complain if no Docker Hub credentials so we aren't bad neighbors
 if [ -z "$DOCKER_HUB_USER" ] || [ -z "$DOCKER_HUB_TOKEN" ]; then
-    echo "WARNING: Docker Hub credentials not set! Falling back to public access which could result in rate limiting."
+    echo "WARNING: Docker Hub credentials not set! Falling back to public access which could result in rate limiting." 1>&2
```
Comment (zjrgov) [inline runner/cf-driver/base.sh:18]: praise: good thinking

File: runner/cf-driver/prepare.sh [hunk 2]
```diff
@@ -51,6 +39,28 @@ get_registry_credentials () {
...
+    local padding="      "
+
+    for v in RUNNER_NAME CACHE_TYPE CACHE_S3_SERVER_ADDRESS CACHE_S3_BUCKET_LOCATION CACHE_S3_BUCKET_NAME CACHE_S3_ACCESS_KEY CACHE_S3_SECRET_KEY; do
+        echo "${padding}${v}: \"${!v}\"" >> "$TMPMANIFEST"
+    done
+
+    # Add any CI_SERVICE_x variables populated by start_service()
+    for v in "${!CI_SERVICE_@}"; do
+        echo "${padding}${v}: \"${!v}\"" >> "$TMPMANIFEST"
+    done
```
Comment (zjrgov) [inline runner/cf-driver/prepare.sh:59]: praise: the indirection is pretty nifty, I haven't used that before

**Review by zjrgov — APPROVED** [review]
I can't remember exactly which thing it is that voids approvals… but hopefully none of these comments will.

File: runner/cf-driver/prepare.sh [hunk 3]
```diff
@@ -51,6 +39,28 @@ get_registry_credentials () {
...
 
+create_temporary_manifest () {
+    # A less leak-prone way to share secrets into the worker which will not
+    # be able to parse VCAP_CONFIGURATION
+    TMPMANIFEST=$(mktemp /tmp/gitlab-runner-worker-manifest.XXXXXXXXXX)
+    chmod 600 "$TMPMANIFEST"
+    cat "${currentDir}/worker-manifest.yml" > "$TMPMANIFEST"
+
+    # Align additional environment variables with YAML at end of source manifest
+    local padding="      "
```
[the remark this answers is not in the document]
Comment (pauldoomgov) [reply runner/cf-driver/prepare.sh:50]: Cool, right? 🧌

**Review by zjrgov — APPROVED** [review]

**Pull request merged**